corgi/core/migrations/0102_fix_duplicate_sbomer_components.py

```diff
@@ -0,0 +1,67 @@
+from django.db import migrations
+
+
+def find_duplicate_sbomer_components(apps, schema_editor) -> None:
+    """Find duplicate quarkus-bom components"""
+    Component = apps.get_model("core", "Component")
+    ComponentNode = apps.get_model("core", "ComponentNode")
+
+    # We create and link nodes using relationships in SBOMer manifests
+    # Maven-type quarkus-bom components are the first in a list of components
+    # They have no nodes or links at all, so they must have no relationships in the manifest
+
+    # Generic-type quarkus-bom components are the root (outside the list) in the manifest
+    # They have nodes and provide all other components
+    # So all the relationships in the manifest must refer to the root component,
+    # AKA the generic-type quarkus-bom component which we artificially create,
+    # and nothing refers to the first component in the list,
+    # AKA the Maven-type quarkus-bom component which is identical to the root
+
+    # Delete the unused / duplicated Maven-type components,
+    # so we can make the root component use Maven type instead
+    # so the root component purls in Corgi match SBOMer
+    # so SDEngine can easily find and manifest the Quarkus components
+    Component.objects.filter(type="MAVEN", name="quarkus-bom").delete()
+
+    for component in Component.objects.filter(type="GENERIC", name="quarkus-bom").iterator():
+        component.type = "MAVEN"
+        if ";" in component.license_declared_raw:
+            component.license_declared_raw = component.license_declared_raw.replace(";", " OR ")
+
+        if "type" not in component.meta_attr:
+            component.meta_attr["type"] = "pom"
+        if "group_id" not in component.meta_attr:
+            component.meta_attr["group_id"] = "com.redhat.quarkus.platform"
+        # TODO: Other qualifiers or meta_attr values?
+
+        bad_purl = component.purl
+        good_purl = bad_purl.replace("pkg:generic/", "pkg:maven/com.redhat.quarkus.platform/", 1)
```

Comment:
All of these components should have purl_declared in meta_attr, if you want to sanity check against that.